tests/datasets/public_dataset_test.py

```diff
@@ -78,3 +80,85 @@ class CustomPublicDataset(PublicDataset):
     assert dataset.root == expected_paths['root']
     assert dataset.path == expected_paths['path']
     assert dataset.downloads_rootpath == expected_paths['download']
+
+
+def test_dataset_download_fails(tmp_path):
+    class CustomPublicDataset(PublicDataset):
+        _mirrors = ['https://www.example.com/', 'https://www.example.com/']
+        _resources = [
+            {
+                'resource': 'v0.4.0.tar.gz',
+                'filename': 'pymovements-0.4.0.tar.gz',
+                'md5': '52bbf03a7c50ee7152ccb9d357c2bb30',
+            },
+        ]
+
+    with mock.patch(
```

Comment:
instead of mocking the private method, just mock `utils.downloads.download_file()`